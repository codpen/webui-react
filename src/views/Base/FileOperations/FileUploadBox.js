import {NativeTypes} from 'react-dnd-html5-backend'
import {useDrop} from 'react-dnd'
import React from "react";
import DropOverlay from "../DropOverlay/DropOverlay";
import * as PropTypes from 'prop-types';


export const FileUploadBox = (props) => {
    const {onDrop} = props
    const [{canDrop, isOver}, drop] = useDrop({
        accept: [NativeTypes.FILE],
        drop(item, monitor) {
            if (onDrop) {
                onDrop(props, monitor)
            }
        },
        collect: (monitor) => ({
            isOver: monitor.isOver(),
            canDrop: monitor.canDrop(),
        }),
    })
    const isActive = canDrop && isOver
    return (
        <div ref={drop}>
            {/*Display overlay if file is about to be dropped*/}
            {isActive && <DropOverlay/>}
            {props.children}
        </div>
    )
}

FileUploadBox.propTypes = {
    onDrop: PropTypes.func
}

export default FileUploadBox;