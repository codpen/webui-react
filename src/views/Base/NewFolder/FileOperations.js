import React from 'react';
import {Button} from "reactstrap";
import NewFolder from "./NewFolder";
import PropTypes from "prop-types";
import Input from "reactstrap/es/Input";
import {connect} from "react-redux";
import {changeGridMode, changeVisibilityFilter} from "../../../actions/explorerStateActions";

class FileOperations extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            newFolderModalIsVisible: false
        };
        this.filterOptions = ["Images", "Pdf", "Videos"];
    }

    openNewFolderModal = () => {
        this.setState({newFolderModalIsVisible: true})
    };

    closeNewFolderModal = () => {
        this.setState({newFolderModalIsVisible: false});
        // this.props.updateHandler();
    };

    handleChangeFilter = (e) => {
        const newFilter = e.target.value;

        this.props.changeVisibilityFilter(this.props.containerID, newFilter);

    };


    handleChangeGridMode = (e) => {
        const gridMode = e.target.value;

        this.props.changeGridMode(this.props.containerID, gridMode);

    };


    render() {

        return (

            <div className="float-right mb-3 mt-1 form-inline">
                <Button color="success" className="mr-1" onClick={this.openNewFolderModal}>New Folder</Button>
                {/*<Button color="success" className="ml-2">Copy from URL</Button>*/}
                {/*<Button color="success" className="ml-2">Copy</Button>*/}
                {/*<Button color="success" className="ml-2">Rename</Button>*/}
                {/*<Button color="success" className="ml-2">Upload</Button>*/}
                <NewFolder containerID={this.props.containerID} isVisible={this.state.newFolderModalIsVisible}
                           closeModal={this.closeNewFolderModal}/>


                <Input type={"select"} onChange={this.handleChangeFilter} value={this.props.visibilityFilter}
                       className="ml-1 mr-1">
                    <option key={0}>None</option>
                    {
                        this.filterOptions.map((item, idx) => {
                            return (<option key={item} value={item}>{item}</option>)
                        })
                    }
                </Input>
                <Input type={"select"} onChange={this.handleChangeGridMode} value={this.props.gridMode}>
                    <option value={"grid"}>Grid</option>
                    <option value={"card"}>Card</option>
                </Input>

            </div>


        );
    }
}

FileOperations.propTypes = {
    updateHandler: PropTypes.func.isRequired,
    containerID: PropTypes.string.isRequired,
    changeVisibilityFilter: PropTypes.func.isRequired,
    visibilityFilter: PropTypes.string,
    gridMode: PropTypes.string
};

const mapStateToProps = (state, ownProps) => ({
    visibilityFilter: state.explorer.visibilityFilters[ownProps.containerID],
    gridMode: state.explorer.gridMode[ownProps.containerID]


});


export default connect(mapStateToProps, {changeVisibilityFilter, changeGridMode})(FileOperations);