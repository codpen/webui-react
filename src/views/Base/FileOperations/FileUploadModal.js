import React, {useCallback, useState} from 'react';
import {
    Button,
    Col,
    Container,
    FormGroup,
    Input,
    Modal,
    ModalBody,
    ModalFooter,
    ModalHeader,
    Row,
    Table,
    UncontrolledTooltip
} from 'reactstrap';
import axiosInstance from "../../../utils/API/API";
import {addColonAtLast, formatBytes} from "../../../utils/Tools";
import {toast} from "react-toastify";
import * as RclonePropTypes from "../../../utils/RclonePropTypes";
import FileUploadBox from "./FileUploadBox";
import {DndProvider} from "react-dnd";
import {HTML5Backend} from "react-dnd-html5-backend";

/**
 * New Mount Modal shows a button for opening a modal for new mount and then executes okHandle when positive
 * button is clicked
 * @param props
 * @returns {*}
 * @constructor
 */
const FileUploadModal = (props) => {
    const {
        currentPath
    } = props;

    const [modal, setModal] = useState(false);


    const [files, setFiles] = useState(null);

    const [isUploading, setIsUploading] = useState(false);


    const toggle = () => setModal(!modal);


    const changeFilesHandler = (e) => {
        setFiles(e.target.files);
    }

    const isUploadDisabled = () => !files || isUploading;

    const getFilesElement = () => {
        let out = [];
        for (let i = 0; i < files.length; i++) {
            out.push(<tr>
                <td>{files[i].name}</td>
                <td>{files[i].size === -1 ? "-" : formatBytes(files[i].size, 2)}</td>
            </tr>);
        }
        return out;
    }


    const fileUploadHandler = (_) => {
        const data = new FormData();
        for (let i = 0; i < files.length; i++) {
            // get item
            let file = files[i];

            data.append("file" + i, file);
        }
        setIsUploading(true);

        axiosInstance.post(`operations/uploadfile?fs=` +
            `${addColonAtLast(currentPath.remoteName)}` +
            `&remote=${currentPath.remotePath}`, data)
            .then(res => {
                setIsUploading(false);
                setFiles(null);
                toggle();
                toast.info("File uploaded successfully");
            }, err => {
                setIsUploading(false);
                toast.error("File upload failed");
            });
    }

    const filesDropHandler = useCallback((item, monitor) => {
        if (monitor) {
            setFiles(monitor.getItem().files);
        }
    }, []);

    return (
        <div data-test="fileUploadModalComponent">

            <Button className="btn-explorer-action" id="UploadButton"
                    onClick={toggle}>
                <i className="fa fa-lg fa-upload"/>
            </Button>
            <UncontrolledTooltip placement="right" target="UploadButton">
                Upload file(s)
            </UncontrolledTooltip>
            <Modal isOpen={modal} toggle={toggle}>
                <DndProvider backend={HTML5Backend}>
                    <ModalHeader toggle={toggle}>New Mount</ModalHeader>
                    <ModalBody>
                        <Container fluid>
                            <FileUploadBox onDrop={filesDropHandler}>
                                {files ? <Row>
                                    <Table>
                                        <thead>
                                        <tr>
                                            <th>Name</th>
                                            <th>Size</th>
                                        </tr>
                                        </thead>
                                        <tbody>
                                        {getFilesElement()}
                                        </tbody>
                                    </Table>

                                </Row> : <Row className="upload-box">
                                    <Col sm={3}/>
                                    <Col sm={6}>

                                        <label htmlFor="upload-file">

                                            <Row>
                                                <i className="fa fa-lg fa-plus mx-auto mt-5"/>
                                                <p className="text-center mt-2">Click here or drag and drop files to
                                                    upload</p>
                                            </Row>

                                        </label>
                                    </Col>
                                    <Col sm={3}/>
                                </Row>
                                }
                                <Row>
                                    <FormGroup row className="pl-2 pr-2 d-none">
                                        <Input id="upload-file" type="file" name="file" onChange={changeFilesHandler}
                                               multiple/>
                                    </FormGroup>
                                </Row>
                            </FileUploadBox>
                        </Container>
                    </ModalBody>
                    <ModalFooter>
                        <Button data-test="ok-button" color="primary" onClick={fileUploadHandler}
                                disabled={isUploadDisabled()}>{isUploading ? "Uploading" : "Upload"}</Button>{' '}
                        <Button data-test="cancel-button" color="secondary" onClick={toggle}>Cancel</Button>
                    </ModalFooter>
                </DndProvider>

            </Modal>
        </div>
    );
}

FileUploadModal.propTypes = {
    currentPath: RclonePropTypes.PROP_CURRENT_PATH
}

export default FileUploadModal;