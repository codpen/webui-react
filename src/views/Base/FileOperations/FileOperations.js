import React from 'react';
import {
	Button,
	ButtonDropdown,
	ButtonGroup,
	Col,
	DropdownItem,
	DropdownMenu,
	DropdownToggle,
	Form,
	FormGroup,
	Input,
	Modal,
	ModalBody,
	ModalFooter,
	ModalHeader,
	Row,
	Spinner,
	UncontrolledTooltip
} from "reactstrap";
import NewFolder from "../NewFolder/NewFolder";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {
	changeGridMode,
	changeVisibilityFilter,
	getFilesForContainerID,
	navigateBack,
	navigateFwd,
	setLoadImages,
	setSearchQuery
} from "../../../actions/explorerStateActions";
import {visibilityFilteringOptions} from "../../../utils/Constants";
import {getAbout} from "../../../actions/providerStatusActions";
import {Doughnut} from "react-chartjs-2";
import {addColonAtLast, bytesToGB, isEmpty, isLocalRemoteName} from "../../../utils/Tools";
import axiosInstance from "../../../utils/API/API";
import {toast} from "react-toastify";
import {PROP_FS_INFO} from "../../../utils/RclonePropTypes";
import urls from "../../../utils/API/endpoint";
import newFolderImg from '../../../assets/img/new-folder.png'; // with import

/**
 * File Operations component which handles user actions for files in the remote.( Visibility, gridmode, back, forward etc)
 */
class FileOperations extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            newFolderModalIsVisible: false,
            isAboutModalOpen: false,
            dropdownOpen: false,
            searchOpen: false
        };
        this.filterOptions = visibilityFilteringOptions;
    }

    openNewFolderModal = () => {
        const {fsInfo} = this.props;
        if (fsInfo && fsInfo.Features && fsInfo.Features.CanHaveEmptyDirectories) {
            this.setState({newFolderModalIsVisible: true});
        } else {
            toast.error("This remote cannot have empty directories");
        }
    };

    closeNewFolderModal = () => {
        this.setState({newFolderModalIsVisible: false});
    };

    handleChangeFilter = (e) => {
        const newFilter = e.target.value;

        const {changeVisibilityFilter} = this.props;

        changeVisibilityFilter(this.props.containerID, newFilter);

    };


    handleChangeGridMode = () => {
        const {gridMode, changeGridMode, containerID} = this.props;
        changeGridMode(containerID, gridMode === "list" ? "card" : "list");
    };

    changeSearch = (e) => {
        e.preventDefault();
        const {containerID} = this.props;
        this.props.setSearchQuery(containerID, e.target.value);
    };


    toggleDropDown = () => {
        this.setState((prevState) => {
            return {
                dropdownOpen: !prevState.dropdownOpen
            }
        })
    };

    toggleAboutModal = () => {
        const {fsInfo} = this.props;
        if (fsInfo && fsInfo.Features && fsInfo.Features.About) {
            this.setState((prevState) => {
                return {
                    isAboutModalOpen: !prevState.isAboutModalOpen
                }
            }, () => {
                if (this.state.isAboutModalOpen) {
                    const {containerID} = this.props;
                    this.props.getAbout(containerID);
                }
            });
        } else {
            toast.error("This remote does not support About");
        }

    };

    handleCleanTrash = () => {
        const {fsInfo} = this.props;
        if (fsInfo && fsInfo.Features && fsInfo.Features.CleanUp) {

            if (window.confirm("Are you sure you want to clear the trash. This operation cannot be undone")) {

                const {currentPath, containerID} = this.props;
                let {remoteName} = currentPath;

                if (!isLocalRemoteName(remoteName)) {
                    remoteName = addColonAtLast(remoteName);
                }

                axiosInstance.post(urls.cleanUpRemote, {
                    fs: remoteName
                }).then((res) => {
                        if (res.status === 200) {
                            toast('Trash Cleaned');
                            this.props.getAbout(containerID);

                        }
                    },
                    (err) => {
                        toast.error("Error clearing trash " + err);
                    }
                )
            }
        } else {
            // Cleanup is not allowed
            toast.error("Clearing trash is not allowed on this remote");
        }
    };

    changeLoadMedia = (e) => {
        e.stopPropagation();
        // console.log(e);
        const {setLoadImages, containerID, loadImages} = this.props;
        setLoadImages(containerID, !loadImages);
    };

	handleSearchOpen = () => {
		const {containerID} = this.props;
		this.setState((prevState) => {
				if (prevState.searchOpen) {
					// Clear Search Query if the search is about to close
					this.props.setSearchQuery(containerID, "");
				}

				return {searchOpen: !prevState.searchOpen}
			}
		);
	};


    render() {
        const {containerID, getFilesForContainerID, gridMode, navigateFwd, navigateBack, searchQuery, currentPath, doughnutData} = this.props;
        const {newFolderModalIsVisible, dropdownOpen, isAboutModalOpen, searchOpen} = this.state;

        const {remoteName, remotePath} = currentPath;

        return (
            <nav aria-label="breadcrumb" className="row mt-3 mb-1">
				<Col sm={4} md={3} className="pl-0">
					<Button color="light" className={"mr-1 btn-explorer-action"}
							onClick={() => navigateBack(containerID)}><i
						className={"fa fa-lg fa-arrow-left"}/></Button>
					<Button color="light" className={"mr-1 btn-explorer-action"}
							onClick={() => navigateFwd(containerID)}><i
						className={"fa fa-lg fa-arrow-right"}/></Button>
					<Button className="mr-1 btn-explorer-action" id="RefreshButton"
							onClick={() => getFilesForContainerID(containerID)}><i
						className={"fa fa-lg fa-repeat"}/></Button>
					<UncontrolledTooltip placement="right" target="RefreshButton">
						Refresh Files
					</UncontrolledTooltip>
				</Col>
				<Col sm={8} md={searchOpen ? 3 : 5}>
					<ol className="breadcrumb float-center" style={{padding: "6px 12px"}}>
						<li className="breadcrumb-item active">{remoteName}:/</li>
						{remotePath}
					</ol>
				</Col>
				<Col sm={12} md={searchOpen ? 6 : 4} className="pr-0">
					<div className="float-right form-inline">

						<ButtonGroup>
							<Form inline>
								<FormGroup>
									{searchOpen && <Input type="text" placeholder="Search" value={searchQuery}
														  className="animate-fade-in"
														  onChange={this.changeSearch}/>
									}
									<Button className="mr-1 btn-explorer-action" onClick={this.handleSearchOpen}>
                                        <i className={"fa fa-lg " + (searchOpen ? "fa-close" : "fa-search")}/>
                                    </Button>
                                </FormGroup>
                            </Form>
                            <Button className="mr-1 btn-explorer-action p-1" id="CreateFolderButton"
                                    onClick={this.openNewFolderModal}><img src={newFolderImg} alt="New Folder" className="fa fa-lg"/> </Button>
                            <UncontrolledTooltip placement="bottom" target="CreateFolderButton">
                                Create a new Folder
                            </UncontrolledTooltip>

                            <ButtonDropdown  isOpen={dropdownOpen} toggle={this.toggleDropDown} direction={'down'} id="FilterButton">
								<DropdownToggle className="btn-explorer-action">
									<i className={"fa fa-lg fa-filter"}/>
								</DropdownToggle>
								<DropdownMenu>
									<DropdownItem key={"None"} value={""}
												  onClick={this.handleChangeFilter}>None</DropdownItem>
									{
										this.filterOptions.map((item, _) => {
											return (<DropdownItem key={item} value={item}
																  onClick={this.handleChangeFilter}>{item}</DropdownItem>)
										})
									}
								</DropdownMenu>
                            </ButtonDropdown>

                            <Button className="btn-explorer-action" id="ListViewButton"
                                    onClick={this.handleChangeGridMode}>
                                <i className={"fa fa-lg " + (gridMode === "card" ? "fa-list" : "fa-th-large")}/>
                            </Button>
                            <UncontrolledTooltip placement="right" target="ListViewButton">
                                {(gridMode === "card" ? "List View" : "Card View")}
                            </UncontrolledTooltip>
                        </ButtonGroup>


                        <NewFolder containerID={containerID} isVisible={newFolderModalIsVisible}
                                   closeModal={this.closeNewFolderModal}/>

                        <Modal isOpen={isAboutModalOpen} toggle={this.toggleAboutModal}>
                            <ModalHeader>
                                Status for {remoteName}
                            </ModalHeader>
                            <ModalBody>
                                <Row>
                                    <Col sm={12}>
                                        <div className="chart-wrapper">
                                            <p>Space Usage (in GB)</p>
                                            {doughnutData && !isEmpty(doughnutData) ? <Doughnut data={doughnutData}/> :
                                                <React.Fragment><Spinner color="primary"/>Loading</React.Fragment>}
                                        </div>
                                    </Col>
                                </Row>
                                <Row>
                                    <Col sm={12}>
                                        <Button color="danger" onClick={this.handleCleanTrash}>Clean Trash <i
                                            className="fa fa-lg fa-trash"/></Button>
                                    </Col>
                                </Row>

                            </ModalBody>
                            <ModalFooter>

                            </ModalFooter>

                        </Modal>

                    </div>
                </Col>
            </nav>

        );
    }
}

FileOperations.propTypes = {
    /**
     * Container ID of the current remote explorer
     */
    containerID: PropTypes.string.isRequired,
    /**
     * Redux function to change the visibility of images/ pdf etc.
     */
    changeVisibilityFilter: PropTypes.func.isRequired,
    /**
     * The current visibility filter setting
     */
    visibilityFilter: PropTypes.string,
    /**
     * Render mode: Grid/Card
     */
    gridMode: PropTypes.string,
    /**
     * Redux function to set the search query as typed by user.
     */
    setSearchQuery: PropTypes.func.isRequired,
    /**
     * Currently set search Query from redux
     */
    searchQuery: PropTypes.string,
    /**
	 * A map which gives the information about the remote about.
	 */
	remoteAbout: PropTypes.object,
	/**
	 * File system information and features about the current remote
	 */
	fsInfo: PROP_FS_INFO,
	/**
	 * Map of data to be passed to the doughnutChart.
	 */
	doughnutData: PropTypes.object,

	numCols: PropTypes.number.isRequired,
};

const mapStateToProps = (state, ownProps) => {
    const remoteAbout = state.providerStatus.about[ownProps.containerID];
    let doughnutData = {};
    const currentPath = state.explorer.currentPaths[ownProps.containerID];
    let fsInfo = {};

    if (currentPath && state.remote.configs && state.remote.configs[currentPath.remoteName]) {
        fsInfo = state.remote.configs[currentPath.remoteName];
    }

    if (remoteAbout) {

        let labels = [];
        let data = [];

        for (const [key, value] of Object.entries(remoteAbout)) {
            if (key !== "total") {
                labels.push(key);
                data.push(bytesToGB(value).toFixed(2));
            }
        }
        if (labels.length > 1 && data.length > 1) {
            doughnutData = {
                labels: labels, datasets: [
                    {
                        data: data,
                        backgroundColor: [
                            '#FF6384',
                            '#36A2EB',
                            '#FFCE56',
                            '#ff7459',
                        ],
                        hoverBackgroundColor: [
                            '#FF6384',
                            '#36A2EB',
                            '#FFCE56',
                            '#ff7459',
                        ],
                    }
                ]
            };
        }
    }

    return {
		visibilityFilter: state.explorer.visibilityFilters[ownProps.containerID],
		loadImages: state.explorer.loadImages[ownProps.containerID],
		currentPath: state.explorer.currentPaths[ownProps.containerID],
		gridMode: state.explorer.gridMode[ownProps.containerID],
		searchQuery: state.explorer.searchQueries[ownProps.containerID],
		fsInfo,
		doughnutData,
		numCols: state.remote.numCols,

	}
};


export default connect(mapStateToProps, {
    changeVisibilityFilter,
    changeGridMode,
    navigateBack,
    navigateFwd,
    getFilesForContainerID,
    setSearchQuery,
    getAbout,
    setLoadImages
})(FileOperations);
