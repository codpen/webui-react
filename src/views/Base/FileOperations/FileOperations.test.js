import React from "react";
import {shallow} from "enzyme";
import {testStore} from "../../../../Utils";
import FileOperations from "./FileOperations";

const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);
    const component = shallow(<FileOperations {...props} store={store}/>);
    return component.childAt(0).dive();
};


describe('File Operations', function () {


    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {

                explorer: {
                    visibilityFilters: {"1": "Images"},
                    gridMode: {
                        "1": "card"
                    },
                    searchQueries: {
                        "1": "abcd"
                    },
                    currentPaths: {
                        "1": {
                            remoteName: "",
                            remotePath: ""
                        }
                    }

                }

            };

            const props = {
                containerID: "1",
                changeVisibilityFilter: jest.fn()
            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            expect(wrapper).toHaveLength(1)
        });
    });
});