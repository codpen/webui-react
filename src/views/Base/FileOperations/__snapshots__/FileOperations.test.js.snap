// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Operations renders should match snapshot 1`] = `
<div
  className="pl-0 mt-3 mb-1 d-flex justify-content-between align-items-center"
  style={
    Object {
      "marginLeft": "-15px",
      "marginRight": "-15px",
    }
  }
>
  <div
    className="d-flex flex-nowrap"
  >
    <Button
      className="mr-1 btn-explorer-action"
      color="light"
      onClick={[Function]}
      tag="button"
    >
      <i
        className="fa fa-lg fa-arrow-left"
      />
    </Button>
    <Button
      className="mr-1 btn-explorer-action"
      color="light"
      onClick={[Function]}
      tag="button"
    >
      <i
        className="fa fa-lg fa-arrow-right"
      />
    </Button>
    <Button
      className="mr-1 btn-explorer-action"
      color="secondary"
      id="RefreshButton"
      onClick={[Function]}
      tag="button"
    >
      <i
        className="fa fa-lg fa-repeat"
      />
    </Button>
    <UncontrolledTooltip
      placement="right"
      target="RefreshButton"
    >
      Refresh Files
    </UncontrolledTooltip>
  </div>
  <div
    className="flex-grow-1 pl-1 pr-1 pr-lg-3 pl-lg-3"
  >
    <Form
      className="h-100"
      inline={true}
      onSubmit={[Function]}
      tag="form"
    >
      <Input
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        type="text"
        value="local:/"
      />
      <Button
        className="d-none"
        color="link"
        style={
          Object {
            "marginLeft": "-45px",
          }
        }
        tag="button"
        type="submit"
      >
        <i
          className="fa fa-arrow-right"
        />
      </Button>
    </Form>
  </div>
  <div
    className="d-flex flex-wrap"
  >
    <Button
      className="mr-1 btn-explorer-action p-1"
      color="secondary"
      id="CreateFolderButton"
      onClick={[Function]}
      tag="button"
    >
      <img
        alt="New Folder"
        className="fa fa-lg"
        src="new-folder.png"
      />
       
    </Button>
    <UncontrolledTooltip
      placement="bottom"
      target="CreateFolderButton"
    >
      Create a new Folder
    </UncontrolledTooltip>
    <ButtonDropdown
      direction="down"
      id="FilterButton"
      isOpen={false}
      toggle={[Function]}
    >
      <DropdownToggle
        aria-haspopup={true}
        className="btn-explorer-action"
        color="secondary"
      >
        <i
          className="fa fa-lg fa-filter"
        />
      </DropdownToggle>
      <DropdownMenu
        flip={true}
        tag="div"
      >
        <DropdownItem
          key="None"
          onClick={[Function]}
          tag="button"
          toggle={true}
          value=""
        >
          None
        </DropdownItem>
        <DropdownItem
          key="Images"
          onClick={[Function]}
          tag="button"
          toggle={true}
          value="Images"
        >
          Images
        </DropdownItem>
        <DropdownItem
          key="Pdf"
          onClick={[Function]}
          tag="button"
          toggle={true}
          value="Pdf"
        >
          Pdf
        </DropdownItem>
        <DropdownItem
          key="Videos"
          onClick={[Function]}
          tag="button"
          toggle={true}
          value="Videos"
        >
          Videos
        </DropdownItem>
      </DropdownMenu>
    </ButtonDropdown>
    <Button
      className="btn-explorer-action"
      color="secondary"
      id="ListViewButton"
      onClick={[Function]}
      tag="button"
    >
      <i
        className="fa fa-lg fa-th-large"
      />
    </Button>
    <UncontrolledTooltip
      placement="right"
      target="ListViewButton"
    >
      Card View
    </UncontrolledTooltip>
    <Button
      className="btn-explorer-action"
      color="secondary"
      id="InfoButton"
      onClick={[Function]}
      tag="button"
    >
      <i
        className="fa fa-lg fa-info"
      />
    </Button>
    <UncontrolledTooltip
      placement="right"
      target="InfoButton"
    >
      Show Remote Info
    </UncontrolledTooltip>
    <Form
      inline={true}
      tag="form"
    >
      <FormGroup
        tag="div"
      >
        <Button
          className="mr-1 btn-explorer-action"
          color=""
          onClick={[Function]}
          tag="button"
        >
          <i
            className="fa fa-lg fa-search"
          />
        </Button>
      </FormGroup>
    </Form>
    <Connect(NewFolder)
      closeModal={[Function]}
      containerID="0"
      isVisible={false}
    />
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      centered={false}
      container="body"
      fade={true}
      isOpen={false}
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      returnFocusAfterClose={true}
      role="dialog"
      scrollable={false}
      toggle={[Function]}
      unmountOnClose={true}
      zIndex={1050}
    >
      <ModalHeader
        charCode={215}
        closeAriaLabel="Close"
        tag="h5"
        wrapTag="div"
      >
        Status for 
        local
      </ModalHeader>
      <ModalBody
        tag="div"
      >
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            sm={12}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="chart-wrapper"
            >
              <p>
                Space Usage (in GB)
              </p>
              <Spinner
                color="primary"
                tag="div"
                type="border"
              >
                Loading...
              </Spinner>
              Loading
            </div>
          </Col>
        </Row>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            sm={12}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              color="danger"
              onClick={[Function]}
              tag="button"
            >
              Clean Trash 
              <i
                className="fa fa-lg fa-trash"
              />
            </Button>
          </Col>
        </Row>
      </ModalBody>
      <ModalFooter
        tag="div"
      />
    </Modal>
  </div>
</div>
`;
